packages/wallet-adapter-core/src/WalletCore.ts

```diff
@@ -200,6 +201,15 @@ export class WalletCore extends EventEmitter<WalletCoreEvents> {
 
         await this.disconnect();
       }
+      if (isdRedirectable()) {
+        // use wallet deep link
+        if (selectedWallet.deepLink) {
+          const url = encodeURIComponent(window.location.href);
+          window.location.href = `${selectedWallet.deepLink}?url=${url}`;
+        } else {
+          return;
+        }
```

Comment:
we actually want more than this. We likely want the deeplink parameter to be something like

`deeplinkForUrl: (url: string) -> string`

The reason for this is the wallet may want to link all the way into the in app browser.


`petra://explore?url=google.com`

For people that don't want to return that, they can just return the base link always. `petra://`

Reply:
interesting. the initial thought was for a wallet to have a `deeplink` prop on its interface where it can be `deeplink=petra://explore` or `deeplink=petra://` and the adapter would concatenate the url to this string.